include/deal.II/multigrid/mg_transfer_global_coarsening.h

```diff
@@ -310,14 +310,34 @@ class MGTwoLevelTransfer<dim, LinearAlgebra::distributed::Vector<Number>>
   mutable LinearAlgebra::distributed::Vector<Number> vec_coarse;
 
   /**
-   * Constraint matrix on coarse level.
+   * Internal vector for performing manual constraint_coarse.distribute().
    */
-  AffineConstraints<Number> constraint_coarse;
+  mutable LinearAlgebra::distributed::Vector<Number> vec_coarse_constraints;
 
   /**
-   * Internal vector that performs constraint_coarse.distribute().
+   * Constraint entries for performing manual constraint_coarse.distribute().
    */
-  mutable LinearAlgebra::distributed::Vector<Number> vec_coarse_constraints;
+  std::vector<std::pair<unsigned int, Number>>
+    constraint_coarse_distribute_entries;
```

Comment:
`AffineConstraints` aside for a moment, I think the canonical way of storing these is to use two separate arrays for the index and the value; at least with `unsigned int` and `double` numbers, you would spend 16 bytes per entry despite only 12 bytes of actual data. However, the most interesting use of this class is probably `float`, so it would fit from that perspective, so I'm leaning towards choosing this version. What do you think?